service/assessment/assessment.go

```diff
@@ -289,11 +282,33 @@ func (svc *Service) AssessEvidences(stream assessment.Assessment_AssessEvidences
 	}
 }
 
-// handleEvidence is the helper method for the actual assessment used by AssessEvidence and AssessEvidences
-func (svc *Service) handleEvidence(ctx context.Context, ev *evidence.Evidence, resourceId string) (results []*assessment.AssessmentResult, err error) {
-	var types []string
+// handleEvidence is the helper method for the actual assessment used by AssessEvidence and AssessEvidences. This will
+// also validate the resource embedded into the evidence and return an error if validation fails. In order to
+// distinguish between internal errors and validation errors, this function already returns a gRPC error.
+func (svc *Service) handleEvidence(ctx context.Context, ev *evidence.Evidence) (results []*assessment.AssessmentResult, err error) {
+	var (
+		types    []string
+		m        proto.Message
+		resource ontology.IsResource
+	)
+
+	// First, try to extract the resource out of the evidence and validate it
+	m, err = ev.Resource.UnmarshalNew()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not unmarshal resource proto message: %v", err)
+	}
```

Comment:
Should we move that to api.Validate()?

Reply:
It is a special case only for the wrapped resource. Adding it to the generic validate might be confusing.

Reply:
Additionally, handling it here has the advantage that we only need to unwrap the resource message once here. It is additionally unwrap in the Rego code, but I am trying to get rid of that as well.